core/src/main/java/io/grpc/ServerStreamTracer.java

```diff
@@ -54,4 +69,116 @@ public void serverCallStarted(ServerCall<?, ?> call) {
     public abstract ServerStreamTracer newServerStreamTracer(
         String fullMethodName, Metadata headers);
   }
+
+  /**
+   * A data class with info about the started {@link ServerCall}.
+   */
+  public static final class ServerCallInfo<ReqT, RespT> {
+    private final MethodDescriptor<ReqT, RespT> methodDescriptor;
+    private final Attributes attributes;
+    private final boolean isReady;
+    private final boolean isCancelled;
+    private final String authority;
+
+    ServerCallInfo(
+        MethodDescriptor<ReqT, RespT> methodDescriptor,
+        Attributes attributes,
+        boolean isReady,
+        boolean isCancelled,
+        String authority) {
+      this.methodDescriptor = methodDescriptor;
+      this.attributes = attributes;
+      this.isReady = isReady;
+      this.isCancelled = isCancelled;
+      this.authority = authority;
+    }
+
+    public MethodDescriptor<ReqT, RespT> getMethodDescriptor() {
+      return methodDescriptor;
+    }
+
+    public Attributes getAttributes() {
+      return attributes;
+    }
+
+    public boolean isReady() {
+      return isReady;
+    }
+
+    public String getAuthority() {
+      return authority;
+    }
+
+    public boolean isCancelled() {
+      return isCancelled;
+    }
+
```

Comment:
ServerCallInfo is passed only to `serverCallStarted()`. `isReady()` and `isCancelled()` don't seem to be useful at that point. We should probably remove them.